leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/impl/RegisterResource.java

```diff
@@ -209,19 +209,27 @@ private void handleUpdate(CoapExchange exchange, Request request, String registr
         String smsNumber = null;
         BindingMode binding = null;
         Link[] objectLinks = null;
+        Map<String, String> additionalParams = new HashMap<>();
+
         for (String param : request.getOptions().getUriQuery()) {
             if (param.startsWith(QUERY_PARAM_LIFETIME)) {
                 lifetime = Long.valueOf(param.substring(3));
             } else if (param.startsWith(QUERY_PARAM_SMS)) {
                 smsNumber = param.substring(4);
             } else if (param.startsWith(QUERY_PARAM_BINDING_MODE)) {
                 binding = BindingMode.valueOf(param.substring(2));
+            } else {
+                String[] tokens = param.split("\\=");
+                if (tokens != null && tokens.length == 2) {
```

Comment:
why is this limited to one parameter?

Reply:
That was the only "kind" of parameters supported for register request. 
If we want to support more than that this should be done in another PR.

@AlexITC which kind of additional Params would you like to support ?

Reply:
What do you mean one parameter? the for loop adds all the additional parameters that come in the update message.

Reply:
my mistake, I just saw the loop.

Reply:
I thought that you was talking about something like "parameter without value".
Total misunderstanding :laughing:  !